xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/objects/classes/DBListClass.java

```diff
@@ -111,7 +112,14 @@ private String toStringButEmptyIfNull(Object object)
     {
         List<ListItem> list = getCachedDBList(context);
         if (list == null) {
-            String hqlQuery = getQuery(context);
+            String hqlQuery = null;
+            try {
+                hqlQuery = getQuery(context);
+            } catch (XWikiException e) {
+                LOGGER.error("Failed to get the list", e);
```

Comment:
What "list" is the error message referring to? The try block is only calling ``getQuery``, so it looks to me that the only thing that can fail there is getting the query.